### ISSUE TITLE
Assert intersection error

### ISSUE THREAD

**Issue by nicolastrres** [pipermerriam/flex#48]
I think we should change the way we validate the types, because this is a valid scenario:

```ssh
properties:
type:
       type: string
       required: True
```
and because of this:
```ssh
# I don't know why this set intersection is enforced...?  Why did I do this.
        intersection = set(schema['properties'].keys()).intersection(schema.keys())
        assert not intersection
```
I am getting the error: assert Intersection.

**Comment by pipermerriam** [comment]
Can you paste me a full code sample.  I'm having trouble reproducing the error with what you provided.

**Comment by pipermerriam** [comment]
Ok, I know why the assertion is in place.

```python
    validators = {}
    if '$ref' in schema:
        validators['$ref'] = LazyReferenceValidator(
            schema['$ref'],
            context,
        )
    if 'properties' in schema:
        # I don't know why this set intersection is enforced...?  Why did I do this.
        intersection = set(schema['properties'].keys()).intersection(schema.keys())
        assert not intersection

        for property_, property_schema in schema['properties'].items():
            property_validators = construct_schema_validators(
                property_schema,
                context,
            )
            validators[property_] = skip_if_empty(skip_if_not_of_type(OBJECT)(
                chain_reduce_partial(
                    operator.methodcaller('get', property_, EMPTY),
                    functools.partial(validate_object, validators=property_validators),
                ),
            ))
```

This function begins by creating a dictionary (`validators`) and then assigning validation function to keys in that dictionary.  The assertion that the property keys do not intersect with the schema keys ensures that we don't overwrite any of the primary schema validators with those from the property validators.  I'm assuming that in your full schema you both define a `type` for your schema as well as it having a `type` property.

This is a shortcoming of using a plain old dictionary to store the validation functions.

This limitation is fixed in https://github.com/pipermerriam/flex/pull/42 which is still in progress.  I've been slowly chipping away at the refactor but it's a big one and is taking time.

**Comment by nicolastrres** [comment]
*I'm assuming that in your full schema you both define a type for your schema as well as it having a type property.* **That is right**, I am using type for the schema and type property.

So this is going to be fixed in #42 :) I am happy to heard that.

Thanks @pipermerriam

**Comment by pipermerriam** [comment]
@nicolastrres I've got all the tests passing in #42 

I'll plan on merging it into master tomorrow and making a release.  Your specific issue was fixed in dfe55b9
